internal/pgdump/extensions.go

```diff
@@ -0,0 +1,63 @@
+package pgdump
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+// PartialCopyWithoutExtensions will perform a partial copy of a SQL database dump from
+// src to dst while commenting out EXTENSIONs-related statements. When it determines there
+// are no more EXTENSIONs-related statements, it will return, resetting src to the position
+// of the last contents written to dst.
+//
+// This is needed for import to Google Cloud Storage, which does not like many EXTENSION
+// statements. For more details, see https://cloud.google.com/sql/docs/postgres/import-export/import-export-dmp
+//
+// Filtering requires reading entire lines into memory - this can be a very expensive
+// operation, so when filtering is complete the more efficient io.Copy should be used
+// to perform the remainder of the copy from src to dst.
+func PartialCopyWithoutExtensions(dst io.Writer, src io.ReadSeeker, progressFn func(int64)) (written int64, err error) {
+	var (
+		reader = bufio.NewReader(src)
+		// position we have consumed up to, track separately because bufio.Reader may have
+		// read ahead on src. This allows us to reset src later.
+		consumed int64
+		// set to true when we have done all our filtering
+		noMoreExtensions bool
+	)
+
+	for !noMoreExtensions {
+		// Read up to a line, keeping track of our position in src
+		var line []byte
+		line, err = reader.ReadBytes('\n')
+		consumed += int64(len(line))
+		if err != nil {
+			return
```

Comment:
Style nit: I was surprised to not see an `err` here or in the rest of the function

Reply:
+1 nit: not a fan of naked return... 😛

Reply:
Updated to use explicit returns! [8020472](https://github.com/sourcegraph/src-cli/pull/893/commits/8020472d04b28ceb8f74b25241eb1179437ae6fb)